masterfabric-mobile/swift-camp
# CL - Canvas Viper Implementation

## 📋 PR Description

This PR introduces the implementation of CanvasView using VIPER architecture. The HomeView has been updated to include a new button that navigates to the CanvasView. On this screen, users can experiment with dynamic shapes (Circle, Rectangle, Ellipse, Capsule) and customize their colors using SwiftUI's Canvas feature. A Picker allows users to switch between shapes, while a ColorPicker lets them choose their preferred color.

---

## ✅ Checklist


---

## 🛠 Steps to Test

1. Launch the application.
2. Navigate to the HomeView.
3. Click the "Go to Canvas" button.
4. Verify that the CanvasView screen appears.
5. Use the Picker to select a shape (Circle, Rectangle, Ellipse, Capsule).
6. Use the ColorPicker to dynamically change the shape's color.
7. Ensure that the shape and color changes are reflected in the Canvas.
8. Click the back button to return to the HomeView.

---

## 🔗 Related Links

- Documentation: https://developer.apple.com/documentation/swiftui/canvas

---

### 📷 Screenshots (Optional)


| ColorPicker    | CanvasView Circle         | CanvasView Rectangle      | CanvasView Ellipse        | CanvasView Capsule        |
|-------------------------------|---------------------------|---------------------------|---------------------------|---------------------------|
| ![Simulator Screenshot - iPhone 16 Pro - 2024-12-29 at 10 25 09](https://github.com/user-attachments/assets/0efdb56e-00c3-4f8f-a186-bffb53364c56) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-29 at 10 25 03](https://github.com/user-attachments/assets/cf2c98f1-c6e5-418c-ba16-dcf03a199e80) |![Simulator Screenshot - iPhone 16 Pro - 2024-12-29 at 10 25 12](https://github.com/user-attachments/assets/d35efa1f-1863-4eac-8119-004f5559efc0) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-29 at 10 25 14](https://github.com/user-attachments/assets/8dc4d245-85e8-47c1-b0fb-3451e2e6a12b)|![Simulator Screenshot - iPhone 16 Pro - 2024-12-29 at 10 25 20](https://github.com/user-attachments/assets/35d49d46-52b4-4d87-9ae0-52e87d319372) |
